kernel/kernel-api/src/main/java/io/delta/kernel/expressions/ScalarExpression.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.expressions;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import static java.util.Objects.requireNonNull;
+
+import io.delta.kernel.expressions.predicates.Predicate;
+
+/**
+ * Scala SQL expressions which takes zero or more inputs and for each input row generates one
+ * output value. See {@link Predicate} for predicate type scalar expressions. Supported
+ * non-predicate type scalar expressions are listed.
+ *  TODO: Currently there aren't any. Will be added in future.
```

Comment:
Can you at least add one inline example to show what kind of SQL functions this is going to capture.